### ISSUE TITLE
Certain non-volatile default values are being flagged in ADD COLUMN statement

### ISSUE THREAD

**Issue by andrewsmith** [sbdchd/squawk#277]
Hello! In linting our migrations, I have noticed that Squawk (master branch) is flagging with `adding-field-with-default` certain ADD COLUMN statements that specify a non-volatile default value. For example:

```
$ squawk --pg-version 13 default.sql 
default.sql:1:0: warning: adding-field-with-default

   1 | ALTER TABLE stale_checks
   2 |   ADD COLUMN cancelled BOOL NOT NULL DEFAULT FALSE;

  note: Adding a field with a VOLATILE DEFAULT requires a table table rewrite with an ACCESS EXCLUSIVE lock. In Postgres versions 11+, non-VOLATILE DEFAULTs can be added without a rewrite.
  help: Add the field as nullable, then set a default, backfill, and remove nullabilty.

find detailed examples and solutions for each rule at https://squawkhq.com/docs/rules
```

The default of `FALSE` is being flagged here, incorrectly in my understanding. I noticed this with several other types as well, including strings, enum types, and jsonb. In my investigation, it looks the tests were only using integer values and integers aren't flagged by this rule.

**Comment by chdsbd** [comment]
Definitely looks like a bug

The `raw_expr` attribute on the `Constraint` has a `TypeCast` wrapper for the boolean values, which integer values don't have. We should add some more test cases and fix this.


```bash
echo 'ALTER TABLE stale_checks ADD COLUMN cancelled BOOL NOT NULL DEFAULT FALSE;' | squawk --dump-ast=raw
```


integer
```json
"raw_expr": {
  "A_Const": {
    "location": 59,
    "val": {
      "Integer": {
        "ival": 1
      }
    }
  }
}
```

boolean
```json
"raw_expr": {
  "TypeCast": {
    "arg": {
      "A_Const": {
        "location": 59,
        "val": {
          "String": {
            "str": "f"
          }
        }
      }
    },
    "location": -1,
    "typeName": {
      "location": -1,
      "names": [
        {
          "String": {
            "str": "pg_catalog"
          }
        },
        {
          "String": {
            "str": "bool"
          }
        }
      ],
      "typemod": -1
    }
  }
}
```
